master/buildbot/reporters/hipchat.py

```diff
@@ -0,0 +1,92 @@
+from buildbot.reporters.http import HttpStatusPushBase
+
+
+class HipChatStatusPush(HttpStatusPushBase):
+    name = "HipChatStatusPush"
+
+    def checkConfig(self, *args, **kwargs):
+        if not isinstance(kwargs.get('auth_token'), basestring):
+            config.error('auth token must be a string')
+        if not isinstance(kwargs.get('endpoint'), basestring):
+            config.error('hipchat endpoint must be a string')
+        if kwargs.get('startCB') and not callable(kwargs['startCB']):
+            config.error("start callback must be callable")
+        if kwargs.get('reviewCB') and not callable(kwargs['reviewCB']):
+            config.error("review callback must be callable")
+        if kwargs.get('summaryCB') and not callable(kwargs['summaryCB']):
+            config.error("summary callback must be callable")
+
+    @defer.inlineCallbacks
+    def reconfigService(self, auth_token, endpoint="https://api.hipchat.com",
+                        startCB=None, startArg=None,
+                        reviewCB=None, reviewArg=None,
+                        summaryCB=None, summaryArg=None,
+                        **kwargs):
+        yield HttpStatusPushBase.reconfigService(self, **kwargs)
+        self.auth_token = auth_token
+        self.endpoint = endpoint
+        self.startCB = startCB
+        self.startArg = startArg
+        self.reviewCB = reviewCB
+        self.reviewArg = reviewArg
+        self.summaryCB = summaryCB
+        self.summaryArg = summaryArg
+        self.user_notify = '%sv2/user/%s/message?auth_token=%s'
+        self.room_notify = '%sv2/room/%s/notification?auth_token=%s'
+
+    @defer.inlineCallbacks
+    def startService(self):
+        yield HttpStatusPushBase.startService(self)
+        startConsuming = self.master.mq.startConsuming
+        self._buildsetCompleteConsumer = yield startConsuming(
+            self.buildsetComplete,
+            ('buildsets', None, 'complete'))
+
+    def stopService(self):
+        self._buildsetCompleteConsumer.stopConsuming()
+        HttpStatusPushBase.stopService(self)
+
+    def buildStarted(self, key, build):
+        return self.getMoreInfoAndSend(build, self.startCB, self.startArg)
+
+    def buildFinished(self, key, build):
+        return self.getMoreInfoAndSend(build, self.reviewCB, self.reviewArg)
+
+    @defer.inlineCallbacks
+    def buildsetComplete(self, key, msg):
+        if self.summaryCB:
+            buildset = yield utils.getDetailsForBuildset(self.master, msg['bsid'], **self.neededDetails)
+            postData = yield self.summaryCB(buildset, self.summaryArg)
+            yield self.send(postData)
+
+    @defer.inlineCallbacks
+    def getMoreInfoAndSend(self, build, callback, callbackArg):
+        if callback:
+            yield utils.getDetailsForBuild(self.master, build, **self.neededDetails)
+            postData = yield callback(build, callbackArg)
+            yield self.send(postData)
+
+    @defer.inlineCallbacks
+    def send(self, params):
```

Comment:
params is not a good name for a variable.

This is changing the send() API from the other http reporters, which is not coherent.
You can change it for all the http reporter (as buildbot nine is not yet released, this is a good time), or not change it at all